### DISCUSSION TITLE
Some additional useful info for "Calculating Liquidity"

### DISCUSSION THREAD

**Discussion by uniyj** [Jeiwan/uniswapv3-book#24] [Milestone 1. First Swap] [answered]
I think it's useful to remind reader that Pb > Pc > Pa, where Pb = price when there is 0 x reserve left (left side of curve); Pc = current price; and Pa = price when there is 0 y reserve left (right side of curve).

Also, per [atiselsts paper](https://atiselsts.github.io/pdfs/uniswap-v3-liquidity-math.pdf), in the text explaining his eq(10), it's useful to think of (Pc, Pb) as the sub-range where x is providing the liquidity (ie segment to the left of Pc), and (Pa, Pc) as the sub-range where y is providing the liquidity (ie segment to the right of Pc).

**Comment by Jeiwan** [comment, marked as answer]
I hope this improves it:
https://github.com/Jeiwan/uniswapv3-book/commit/ed077b0b9b8e905d5c3cbef232f7d5aa9f4bf52d
